src/test/java/org/takes/rq/RqMultipartTest.java

```diff
@@ -42,6 +44,129 @@
+     */
+    @Test
+    public void equalsReturnsTrue() throws IOException {
+        final Request req = new RqFake(
+            Arrays.asList(
+                "POST /h?a=3 HTTP/1.1",
+                "Host: asw.example.com",
+                "Content-Type: multipart/form-data; boundary=AaB01x",
+                "Content-Length: 100008"
+            ),
+            Joiner.on(CR).join(
+                "--AaB01x",
+                "Content-Disposition: form-data; name=\"addres\"",
+                "",
+                "449 N Wolfe Rd, Sunnyvale, CA 94085",
+                "--AaB01x",
+                // @checkstyle LineLength (1 line)
+                "Content-Disposition: form-data; name=\"data\"; filename=\"a.bin\"",
+                "Content-Transfer-Encoding: uyf-8",
+                "",
+                "\r\t\n\u20ac\n\n\n\t\r\t\n\n\n\r\nthe end ",
+                "--AaB01x--"
+            )
+        );
+        Assert.assertEquals(new RqMultipart(req), new RqMultipart(req));
+    }
+
+    /**
+     * RqMultipart throws IOException on missing closing boundary.
+     * @throws IOException if some problem inside
+     */
+    @Test(expected = IOException.class)
+    public void throwsExceptionOnNoClosingBounaryFound() throws IOException {
+        final Request req = new RqFake(
+            Arrays.asList(
+                "POST /h?a=4 HTTP/1.1",
+                "Host: rtw.example.com",
+                "Content-Type: multipart/form-data; boundary=AaB01x",
+                "Content-Length: 100007"
+            ),
+            Joiner.on(CR).join(
+                "--AaB01x",
+                "Content-Disposition: form-data; fake=\"address\"",
+                "",
+                "447 N Wolfe Rd, Sunnyvale, CA 94085",
+                "Content-Transfer-Encoding: uwf-8"
+            )
+        );
+        new RqMultipart(req);
+    }
+
+    /**
+     * RqMultipart throws IOException on missing name
+     * at Content-Disposition header.
+     * @throws IOException if some problem inside
+     */
+    @Test(expected = IOException.class)
+    public void throwsExceptionOnNameNotFound() throws IOException {
+        final Request req = new RqFake(
+            Arrays.asList(
+                "POST /h?a=5 HTTP/1.1",
+                "Host: wwy.example.com",
+                "Content-Type: multipart/form-data; boundary=AaB00x",
+                "Content-Length: 100006"
+            ),
+            Joiner.on(CR).join(
+                "--AaB00x",
+                "Content-Disposition: form-data; fake=\"address\"",
+                "",
+                "340 N Wolfe Rd, Sunnyvale, CA 94085",
+                "--AaB00x",
+                "Content-Transfer-Encoding: rtf-8",
+                "",
+                "\r\t\n\u20ac\n\n\n\t\r\t\n\n\n\r\nthe  end",
+                "--AaB00x--"
+            )
+        );
+        new RqMultipart(req);
+    }
+
+    /**
+     * RqMultipart throws IOException on missing boundary Content-Type header.
```

Comment:
@ekondrashev `can throw` / see http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names for test name and test description.